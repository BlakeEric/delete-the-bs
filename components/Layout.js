// This will wrap the other components and hold the application state to pass down to
// other components.
import Footer from './Footer'
import Header from './Header'

const Layout = (props) => {
  return (
    <>
      <Header />
      <div className="pageInner">
        {props.children}
      </div>
      <Footer />
    </>
  )
}

export default Layout
