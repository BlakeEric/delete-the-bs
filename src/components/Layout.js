// This will wrap the other components and hold the application state to pass down to
// other components.
import Footer from './Footer'
import Header from './Header'

const Layout = (props) => {
  return (
    <>
      <Header />
      <main className="pageInner">
        {props.children}
      </main>
      <Footer />
    </>
  )
}

export default Layout
