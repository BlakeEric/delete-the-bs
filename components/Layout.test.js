/* eslint-env jest */
import ReactDOM from 'react-dom'
import Layout from './Layout'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Layout component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Layout><h1>Child Node</h1></Layout>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a ui as expected, including child components', () => {
    const wrapper = mount(<Layout><h1>Child Node</h1></Layout>)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
