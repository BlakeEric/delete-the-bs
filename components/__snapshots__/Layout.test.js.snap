// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders a ui as expected, including child components 1`] = `
<Layout>
  <Header>
    <header
      aria-label="Credits"
      className="siteHeader"
      id="header"
    >
      <a
        href="http://localhost:3000"
      >
        (-bs)
      </a>
      <Link
        href="/faq"
      >
        <a
          className="faq-link"
          href="/faq"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          FAQ
        </a>
      </Link>
    </header>
  </Header>
  <div
    className="pageInner"
  >
    <h1>
      Child Node
    </h1>
  </div>
  <Footer>
    <footer
      aria-label="Credits"
      className="siteFooter"
      id="footer"
      role="contentinfo"
    >
      <hr
        className="global-divider"
      />
      <p>
        <span>
          Â© Blake Lundquist 2020
        </span>
        <a
          href="https://github.com/BlakeEric/delete-the-bullshit"
          rel="nofollow"
          target="_blank"
        >
          View source code for this website
        </a>
      </p>
    </footer>
  </Footer>
</Layout>
`;
